src/AutoRest.CSharp/Mgmt/Output/ResourceOperation.cs

```diff
@@ -68,24 +72,185 @@ internal class ResourceOperation : TypeProvider
 
         public IDictionary<OperationGroup, MgmtNonResourceOperation> ChildOperations => _childOperations;
 
-        public IEnumerable<PagingMethod> PagingMethods => _pagingMethods ??= ClientBuilder.BuildPagingMethods(OperationGroup, RestClient, Declaration, OverridePagingMethodName);
+        public IEnumerable<PagingMethod> PagingMethods => _pagingMethods ??= ClientBuilder.BuildPagingMethods(OperationGroup, RestClient, Declaration);
 
-        private string OverridePagingMethodName(OperationGroup operationGroup, Operation operation, RestClientMethod restClientMethod)
+        public IEnumerable<ResourceListMethod> ResourceOperationsListMethods => _resourceOperationsListMethods ??= GetResourceOperationsListMethod();
+
+        public ResourceListMethod? SubscriptionExtensionsListMethod => _subscriptionExtensionsListMethod ??= GetSubscriptionExtensionsListMethod();
+
+        public ResourceListMethod? TenantListMethod => _tenantListMethod ??= GetTenantListMethod();
+
+        public ResourceListMethod? LocationsListMethod => _locationsListMethod ??= GetLocationsListMethod();
+
+        public virtual ClientMethod? GetMethod => _getMethod ??= Methods.FirstOrDefault(m => m.Name.StartsWith("Get") && m.RestClientMethod.Responses[0].ResponseBody?.Type.Name == ResourceData.Type.Name);
+
+        protected virtual IEnumerable<ClientMethod> GetMethodsInScope()
         {
-            // override the name for ListBySubscription
-            if (restClientMethod.Name == "ListAll" || restClientMethod.Name == "ListBySubscription")
+            return ClientBuilder.BuildMethods(OperationGroup, RestClient, Declaration);
+        }
+
+        private IEnumerable<ResourceListMethod> GetResourceOperationsListMethod()
+        {
+            return GetListMethods(true, false);
+        }
+
+        private ResourceListMethod? GetSubscriptionExtensionsListMethod()
+        {
+            var listMethods = GetListMethods(false, true);
+            foreach (var listMethod in listMethods)
             {
-                return $"List{ResourceName.ToPlural()}";
+                var restClientMethod = GetRestClientMethod(listMethod);
+                var pathSegments = restClientMethod?.Request.PathSegments;
+
+                // subscriptions scope
+                if (pathSegments.Any(p => p.Value.IsConstant && p.Value.Constant.Value?.ToString() == "/subscriptions/") &&
+                    !pathSegments.Any(p => p.Value.IsConstant && (p.Value.Constant.Value?.ToString() == "/resourceGroups/" || p.Value.Constant.Value?.ToString() == "/providers/Microsoft.Compute/locations/")))
+                {
+                    return listMethod;
+                }
```

Comment:
It will look more clear if we could have some extension method to do something like this

Reply:
I've added four methods for each extensions(Subscription, Resource Group, Tenant, Management Group) to get the List methods for the clarity and added conditions as per the extensions.